src/lib/util.js

```diff
@@ -111,6 +113,65 @@ function createDownloadIframe() {
     return iframe;
 }
 
+/**
+ * Checks if the url is a download host, but not the default download host.
+ *
+ * @public
+ * @param {string} url - The URL to check
+ * @return {boolean} - HTTP response
+ */
+export function isNonDefaultDownloadHost(url) {
+    return !url.startsWith(DEFAULT_DOWNLOAD_HOST_PREFIX) && url.match(/^https:\/\/dl\d+\./);
+}
+
+/**
+ * Replaces the hostname of a download URL with the default hostname, https://dl.
+ *
+ * @public
+ * @param {string} url - The URL to modify
+ * @return {string} - The updated download URL
+ */
+export function replaceDownloadHostWithDefault(url) {
+    return url.replace(/^https:\/\/dl\d+\./, DEFAULT_DOWNLOAD_HOST_PREFIX);
+}
+
+/**
+ * Sets local storage to use the default download host
+ *
+ * @public
+ * @return {void}
+ */
+export function setDownloadHostFallback() {
+    sessionStorage.setItem('download_host_fallback', 'true');
+}
+
+/**
+ * Determines if the degraded download notification should be shown.
+ *
+ * @public
+ * @return {boolean} Should the notification be shown
+ */
+export function shouldShowDegradedDownloadNotification() {
```

Comment:
We will only show the download notification if:
1. we don't have the cookie already
2. our downloads are degraded.

This means at most we can see the notification once per week, and only if we are still in a degraded state.

These will be constants in the final PR :)